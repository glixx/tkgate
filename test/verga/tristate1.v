module top;
  reg a1, b1;
  reg a0, b0;
  reg x;
  integer i;


  assign x = a1 ? a0 : 1'bz;
  assign x = b1 ? b0 : 1'bz;

  initial
    begin
	$display("Starting tri-state");
        for (i = 0;i < 16;i = i + 1)
          begin
	    a0 = (i & 8'h1) != 0;
	    a1 = (i & 8'h2) != 0;
	    b0 = (i & 8'h4) != 0;
	    b1 = (i & 8'h8) != 0;
            #1 $display("%b = buf(%b)if(%b) ++  buf(%b)if(%b)",x,a0,a1,b0,b1);
          end
    end
endmodule
