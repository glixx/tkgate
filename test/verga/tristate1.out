Starting tri-state
z = buf(0)if(0) ++  buf(0)if(0)
z = buf(1)if(0) ++  buf(0)if(0)
0 = buf(0)if(1) ++  buf(0)if(0)
1 = buf(1)if(1) ++  buf(0)if(0)
z = buf(0)if(0) ++  buf(1)if(0)
z = buf(1)if(0) ++  buf(1)if(0)
0 = buf(0)if(1) ++  buf(1)if(0)
1 = buf(1)if(1) ++  buf(1)if(0)
0 = buf(0)if(0) ++  buf(0)if(1)
0 = buf(1)if(0) ++  buf(0)if(1)
0 = buf(0)if(1) ++  buf(0)if(1)
x = buf(1)if(1) ++  buf(0)if(1)
1 = buf(0)if(0) ++  buf(1)if(1)
1 = buf(1)if(0) ++  buf(1)if(1)
x = buf(0)if(1) ++  buf(1)if(1)
1 = buf(1)if(1) ++  buf(1)if(1)
